src/Corax/Pipeline/Parsing/VectorParsers.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+
+namespace Corax.Pipeline.Parsing
+{
+    internal static class VectorParsers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int FindFirstNonAsciiSse(ReadOnlySpan<byte> buffer)
+        {
+            ref byte bufferStart = ref MemoryMarshal.GetReference(buffer);
+
+            nint pos = 0;
+            nint len = buffer.Length;
+
+            // process in blocks of 16 bytes when possible
+            for (; pos + 16 < len; )
+            {
+                Vector128<byte> v1 = Vector128.LoadUnsafe(ref Unsafe.AddByteOffset(ref bufferStart, pos));
+                Vector128<byte> v2 = Vector128.LoadUnsafe(ref Unsafe.AddByteOffset(ref bufferStart, pos + Vector128<byte>.Count));
+                var v = Vector128.BitwiseOr(v1, v2);
+
+                Vector128<byte> mask = Vector128.Create((byte)0x80);
+                Vector128<byte> result = Vector128.BitwiseAnd(v, mask);
+
+                if (Sse41.TestZ(result, result) == false)
```

Comment:
Why not ` result != Vector128<int>.Zero` ? 
Then you don't need and `Sse41` instructions and it would work on ARM as well, no?

Reply:
Changed to use that, generated output is still quite good.